lib/std/hash_map.zig

```diff
@@ -1294,7 +1301,7 @@ pub fn HashMapUnmanaged(
 
             var first_tombstone_idx: usize = self.capacity(); // invalid index
             var metadata = self.metadata.? + idx;
-            while ((metadata[0].isUsed() or metadata[0].isTombstone()) and limit != 0) {
+            while (!metadata[0].isFree() and limit != 0) {
```

Comment:
You might be able to get even more improvement by moving the isFree() check to the end of the loop, so the common case of hitting the first item skips this check.

Reply:
Thanks for the suggestion! I just tested it out here: https://github.com/ziglang/zig/commit/bb49aa09b77f30c5c893fa11d4f488d0c20cf492.

The results are in the spreadsheet.. it's slower, but I don't have a guess why.

Reply:
Interesting, thanks!